Create xref:application-directory-application.adoc (#1723)
bonitasoft/bonita-doc: modules/ROOT/pages/application-directory.adoc

= Bonita Application Directory 
:description: This pages describes the Bonita _Application Directory_: its values, how to access it, and how to customize it. +

{description}

== Values

On the developer side, this application embeds error pages for 403 (forbidden), 404 (not found), 500 (internal server) errors. +

> Those pages will be displayed for all applications, provided or custom, given such errors occur at the level of the URL. +

Reviewer: ```suggestion
Those pages will be displayed for all applications, provided or custom, given such errors occur when trying to display a page. +
```

On the user side, the Application Directory displays all applications that the logged user has access to. +
The access rights are managed through xref:profiles-overview.adoc[profiles]. +
Since a user can be mapped to multiple profiles, and since each profile can be mapped to multiple applications, all users need a place to centralize all Bonita applications they can access. +

This application is diplayed when a user:

> * accesses the root of Bonita (/bonita or *ITS CLOUD EQUIVALENT*)

Reviewer: ```suggestion
* accesses the root of Bonita web application (`/bonita` or `/` if context root is configured to target Bonita web application)
```

When a user uses valid URLs for Bonita, they can always access the list of their applications by using the mosaic icon to switch from the current to another application. +

== Access during Development and Testing phases

To access the application:

. Open Bonita Studio
. Click on the *Applications* icon of the coolbar

You will be logged in as the default user of the deployed organization. This default user can be changed by opening the organization editor. +
By default, in our provided ACME organization, the default user is Walter Bates, logged as "walter.bates", with password "bpm". +
The default application is the Application Directory.

To access its content: 

. From the Application Directory, click on the Bonita Administrator application (also accessible as Walter Bates)
. In the menu, click on "Applications"
. On the "Bonita Application Directory" row, click on the "..." _View application details_ icon 

== Status
This application is core to Bonita Platform. It is embedded in the Runtime and cannot be downloaded from the Studio nor be part of your automation project. +
This is the reason why you will not see it in the studio Project Explorer. +

> Also, since it looks more like a page than an application, it does not use the "Bonita layout" but "Layout Without Menu".

Reviewer: ```suggestion
Also, since it looks more like a page than an application, it does not use the "Bonita layout" but a "Layout Without Menu".
```

It is also the reason why the list of pages is fixed, and no navigation can be created. +

However, it is customizable in quite a few ways, as explained below.

== Customization 

On the application details page, you will be able to edit:

* the logo
* the layout
* the theme

> * each of the error pages

Reviewer: ```suggestion
* each one of the error pages
```

To update the logo:

> . Create the new one. Supported MIME types: XXX. Maximum size: XXX

Reviewer: ```suggestion
. Create the new one. Supported formats/extensions are: png, jpg, jpeg, gif, bmp, wbmp, tga. Maximum size: 100 KB
```

. Click on the "Upload new picture" button below the current logo
. Select the new logo
The new logo is applied.

To update the layout and/or theme:
. Create a new xref:layout-development.adoc[layout] and/or xref:customize-living-application-theme.adoc[theme]
. Go the _Resources_ tab
. Install it 
. Go back to the application details
. Click on the current layout or theme
. Select the new one from the drop down list

The new layout or theme is applied.

> The eroor pages have been created with the UI Designer. To edit them, you need to open them in the UI Designer:

Reviewer: ```suggestion
The error pages have been created with the UI Designer. To edit them, you need to open them in the UI Designer:
```
